R/spark_compile.R

```diff
@@ -45,11 +45,13 @@ spark_compile <- function(jar_name,
   jar_path <- file.path(java_path, jar_name)
 
   scala_path <- file.path(root, "java")
-  scala_files <- list.files(scala_path, pattern = "scala$", full.names = TRUE)
+  scala_files <- list.files(scala_path, pattern = "scala$",
+                            full.names = TRUE,
+                            recursive = TRUE)
 
   # apply user filter to scala files
   if (is.function(filter))
-    scala_files <- filter(scala_files)
+    scala_files <- Filter(filter, scala_files)
```

Comment:
Is there a reason why you want to use `Filter` here? My thought is that the filter function here would take a vector of scala files, and return a new vector of scala files. It seems like we don't really gain much by hiding the vectorization.